### PR TITLE
virus_scan_s3_bucket: multithread main loop

### PR DESCRIPTION
**Pull request by risicle**
This PR AKA "how to multithread a command-line script". Not that I'm being opinionated or anything.

https://trello.com/c/aWnL8fAa/192-antivirus-create-catch-up-bulk-scan-script

Importantly retaining the flexibility to use a regular `map` implementation to disable any threading complexity.

### CODE REVIEW

File: tests/test_virus_scan_s3_bucket.py [hunk 1]
```diff
@@ -135,87 +142,114 @@ def _get_mock_clients(self, versions_responses, versions_page_size):
...
+            assert retval == (7, 0, 0, 0,)
+        else:
+            assert av_api_client.mock_calls == [
+                mock.call.scan_and_tag_s3_object("spade", "sandman/4321-billy-winks.pdf", "oo_.BepoodlLml"),
+                mock.call.scan_and_tag_s3_object("spade", "sandman/4321-billy-winks.pdf", "moB_eLplool.do"),
+                mock.call.scan_and_tag_s3_object("spade", "sandman/4321-billy-winks.pdf", "ooBmo_pe.ldoLl"),
+                mock.call.scan_and_tag_s3_object("spade", "sandman/1234-deedaw.pdf", "epmlLoBodo_ol."),
+                mock.call.scan_and_tag_s3_object("spade", "sandman/1234-deedaw.pdf", "loleLoooBp_md."),
+                mock.call.scan_and_tag_s3_object("spade", "sandman/4321-billy-winks.pdf", "molo.oB_oLdelp"),
+                mock.call.scan_and_tag_s3_object("spade", "dribbling/bib.jpeg", "ldmoo_.pBeolLo"),
```
Comment (katstevens) [inline tests/test_virus_scan_s3_bucket.py]: I'm curious... do the test threads run deterministically?

Comment (risicle) [reply tests/test_virus_scan_s3_bucket.py]: Well caught - I had forgotten to put the re-sorting in. Have now. Think it was just passing out of luck.

File: dmscripts/virus_scan_s3_bucket.py [hunk 2]
```diff
@@ -50,4 +57,28 @@ def virus_scan_bucket(s3_client, antivirus_api_client, bucket_name, prefix="", s
 
             logger.info("%s: %s", version["VersionId"], message)
 
+    try:
+        for _ in map_callable(
```
[the remark this answers is not in the document]
Comment (risicle) [reply dmscripts/virus_scan_s3_bucket.py]: Agree.

File: dmscripts/virus_scan_s3_bucket.py [hunk 3]
```diff
@@ -50,4 +57,28 @@ def virus_scan_bucket(s3_client, antivirus_api_client, bucket_name, prefix="", s
...
+            # about the result
+            pass
+    except BaseException as e:
+        logger.warning(
+            "Aborting with candidate_count = %s, pass_count = %s, fail_count = %s, already_tagged_count = %s",
+            candidate_count,
+            pass_count,
+            fail_count,
+            already_tagged_count,
+        )
```
[the remark this answers is not in the document]
Comment (risicle) [reply dmscripts/virus_scan_s3_bucket.py]: That stuff's really hard to test, man...

**Pull request merged**